cron.go

```diff
@@ -166,13 +171,29 @@ func (c *Cron) Schedule(schedule Schedule, cmd Job) EntryID {
 		Job:        cmd,
 	}
 	if !c.running {
-		c.entries = append(c.entries, entry)
+		c.addNewEntry(entry)
 	} else {
 		c.add <- entry
 	}
 	return entry.ID
 }
 
+func (c *Cron) UpdateSchedule(id EntryID, spec string) error {
+	c.runningMu.Lock()
+	defer c.runningMu.Unlock()
+	schedule, err := c.parser.Parse(spec)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("could not update schedule for job %d", id))
+	}
+	if entry, ok := c.entryLookupTable[id]; ok {
+		entry.Schedule = schedule
+		// Updating entry.Next as it might now be stale.
+		entry.Next = entry.Schedule.Next(c.now())
```

Comment:
This doesn't work -- you can verify that by adding a job with a schedule that doesn't fire for a long time, and then update it to fire immediately. It will not be run. If cron is running already, it is blocked until the next time a schedule was active, expected to be at the beginning of the entries slice. That has to be recalculated when an update happens.

I think that also points to gaps in the test coverage.

Reply:
Good catch :) I have now made sure that the entries are reordered and that the timer is reinitialized so that the updated schedule is accommodated.